server/services/user.js

```diff
@@ -0,0 +1,172 @@
+//  Defines various services for the user object
+
+const Sequelize = require('sequelize');
+const sequelize = require('../../settings/connect');
+const User = require('../../app/models/user')(sequelize, Sequelize);
+const jwt = require('jsonwebtoken');
+const bcrypt = require('bcrypt-nodejs');
+
+const secret = 'documentmanagement'; // Specify a secret to sign json web tokens
+
+/**
+ * Create a new user
+ * @param {object} req
+ * @param {function} res // Callback
+ * @returns {boolean} true if created, false otherwise.
+ */
+module.exports.createUser = (req, res) => {
+  const newUser = req.body;
+  const password = bcrypt.hashSync(newUser.password);
+  User.findOrCreate({
+    where: {
+      email: newUser.email
+    },
+    defaults: {
+      username: newUser.username,
+      name: newUser.name,
+      email: newUser.email,
+      password,
+      roleId: newUser.roleId
+    }
+  })
+    .spread((user, created) => { // spread operator used for returning multiple arguments.
+      if (created) {
+        const token = jwt.sign({
+          userId: user.id,
+          userName: user.username,
+          userRoleId: user.roleId
+        }, secret, { expiresIn: '1 day' });
+        return res.status(201).send({ user, userToken: token, message: 'New User Created! Token expires in a day.' });
+      }
+      return res.status(400).send({ message: 'User already exists' });
+    });
+};
+
+/**
+ * User login
+ * @param {object} req
+ * @param {function} res // Callback
+ * @returns {promise} http response.
+ */
+module.exports.login = (req, res) => {
+  if (!req.query.username || !req.query.password) {
+    return res.status(400).send({ message: 'Invalid request, specify username and password' });
+  }
+  User.find({
+    where: {
+      username: req.query.username
+    }
+  })
+  .then((data) => {
+    if (!data) {
+      return res.status(404).send({ message: 'User not found' });
+    }
+    if (!bcrypt.compareSync(req.query.password, data.password)) {
+      return res.status(400).send({ message: 'Invalid password' });
```

Comment:
Stick to the same `User not found` error so that malicious users won't get a sense of accomplishment when they get the username correctly